pkgs/build-support/trivial-builders.nix

```diff
@@ -249,6 +250,60 @@ rec {
       '';
     };
 
+  /*
+   * Similar to writeShellScriptBin and writeScriptBin.
+   * Writes an executable Shell script to /nix/store/<store path>/bin/<name> and
+   * checks its syntax with shellcheck and the shell's -n option.
+   * Automatically includes sane set of shellopts (errexit, nounset, pipefail)
+   * and handles creation of PATH based on runtimeInputs
+   *
+   * Note that the checkPhase uses stdenv.shell for the test run of the script,
+   * while the generated shebang uses runtimeShell. If, for whatever reason,
+   * those were to mismatch you might lose fidelity in the default checks.
+   *
+   * Example:
+   * # Writes my-file to /nix/store/<store path>/bin/my-file and makes executable.
+   * writeShellApplication {
+   *   name = "my-file";
+   *   runtimeInputs = [ curl w3m ];
+   *   text = ''
+   *     curl -s 'https://nixos.org' | w3m -dump -T text/html
+   *    '';
+   * }
+  */
+  writeShellApplication =
+    { name
+    , text
+    , runtimeInputs ? [ ]
+    , checkPhase ? null
+    }:
+    writeTextFile {
+      inherit name;
+      executable = true;
+      destination = "/bin/${name}";
+      text = ''
+        #!${runtimeShell}
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+
+        export PATH="${lib.makeBinPath runtimeInputs}:$PATH"
```

Comment:
Why prefix the PATH?
Isn't our goal to have pure scripts and therefore the PATH should be replaced?
Prefixing the PATH incentivizes the script to have undefined runtime deps. It's very likely that authors forget to specify dependencies, because the script will just work on their machine.

Reply:
@DavHau This doesn't address your question about appending here, but if you have an acute need: you're describing why resholve exists. Here's the informal nixpkgs doc for [`resholve.writeScript` and `resholve.writeScriptBin`](https://github.com/NixOS/nixpkgs/blob/163f66a804c6b682bb8655532589088cf43326fe/pkgs/development/misc/resholve/README.md#basic-resholvewritescript-and-resholvewritescriptbin-examples)